### PR TITLE
search: Use labeled code fences and add CSS for highlighting

### PR DESCRIPTION
**Pull request by camdencheek**
This PR paves the way for enabling query syntax highlighting on our docs page. 

It does two things:
1) Adds CSS for highlighting `.chroma.sgquery` code blocks with colors that match the query editor
2) Converts our search examples page to use code blocks so that we can take advantage of the highlighting

# How it works
- `docsite` [PR #70](https://github.com/sourcegraph/docsite/pull/70) adds a lexer for our queries, and modifies the generated `<pre>` block surrounding code to include a class named the same as the code block language. For example, for the following code block, the tag would look like `<pre class="chroma sgquery">`

````md
```sgquery
repo:test (a or b)
```
````

- New CSS selectors are added that translate the syntax tags added by chroma to the colors we use in our query editor. These override the colors defined for all other languages. This is so our syntax highlighting for other languages is not affected by this change. 

I am a complete CSS noob, so please take a close look at the CSS and feel free to suggest any modifications. 

# What it looks like:
## Before
before

## After
after

### CODE REVIEW

File: doc/_resources/assets/docsite.css [hunk 1]
```diff
@@ -875,6 +887,18 @@ body.theme-dark .markdown-body .chroma .vi { color: #f8f8f2 } /* Name.Variable.I
 body.theme-dark .markdown-body .chroma .vm { color: #f8f8f2 } /* Name.Variable.Magic */
 body.theme-dark .markdown-body .chroma .il { color: #ae81ff } /* Literal.Number.Integer.Long */
 
+/* Dark query colors */
+body.theme-dark .markdown-body .chroma.sgquery .c { color: #ffa94d } /* Comment */
+body.theme-dark .markdown-body .chroma.sgquery .k { color: #da77f2; font-weight: bold } /* Keyword */
+body.theme-dark .markdown-body .chroma.sgquery .nb { color: #569cd6 } /* Name.Builtin */
+body.theme-dark .markdown-body .chroma.sgquery .nf { color: #da77f2; font-weight: bold } /* Name.Function */
+body.theme-dark .markdown-body .chroma.sgquery .se { color: #ffa8a8; background-color: transparent } /* Literal.String.Escape */
```
Comment (camdencheek) [inline doc/_resources/assets/docsite.css:895]: Is there a better way of ignoring the background color that's inherited on these other than setting `background-color: transparent`?

Comment (limitedmage) [reply doc/_resources/assets/docsite.css:895]: What happens if you don't set the background color to transparent?

Comment (camdencheek) [reply doc/_resources/assets/docsite.css:895]: It inherits the light red background color from line 875

Comment (limitedmage) [reply doc/_resources/assets/docsite.css:895]: I think it's fine to have the transparent override in this case.

File: doc/_resources/assets/docsite.css [hunk 2]
```diff
@@ -875,6 +887,18 @@ body.theme-dark .markdown-body .chroma .vi { color: #f8f8f2 } /* Name.Variable.I
...
+body.theme-dark .markdown-body .chroma.sgquery .sr { color: #ff6b6b; background-color: transparent } /* Literal.String.Regex */
+body.theme-dark .markdown-body .chroma.sgquery .dl { color: #ffa94d; background-color: transparent } /* Literal.String.Delimiter */
+body.theme-dark .markdown-body .chroma.sgquery .sa { color: #3bc9db } /* Literal.String.Affix */
+body.theme-dark .markdown-body .chroma.sgquery .ss { color: #ff6b6b; background-color: transparent } /* Literal.String.Symbol */
```
Comment (limitedmage) [inline doc/_resources/assets/docsite.css:899]: I wonder if we could get these colors from our `colors.scss`? We already have them duplicated in both Monaco settings and SCSS, would rather not have a third copy of these colors.
Vars we have there are are `--search-filter-keyword-color` for filters and  `--search-keyword-color` for and/or.

Comment (camdencheek) [reply doc/_resources/assets/docsite.css:899]: Nice -- I'll give it a try!

**Review by rvantonder — APPROVED** [review]
This is a great improvement with a small footprint so I will just indiscriminately stamp things.

**Comment by camdencheek** [comment]
Added some variables and left a comment on them with a link to the issue about `docsite.css -> docsite.scss`.

**Pull request merged**